src/pages/dashboard/Dashboard.tsx

```diff
@@ -50,43 +49,42 @@ function Dashboard() {
 
   const vaults = getVaults();
 
-  const wrappedCurrencies = useMemo(() => {
-    const currencies = vaults
-      .map((vault) => {
-        return vault.id.currencies.wrapped;
-      })
-      .filter((asset) => {
-        return asset != null;
-      });
+  const availableCurrencies = useMemo(() => {
+    const currencies: SpacewalkPrimitivesCurrencyId[] = [];
+    vaults.forEach((vault) => {
+      currencies.push(vault.id.currencies.wrapped);
+      currencies.push(vault.id.currencies.collateral);
+    });
 
     // Deduplicate assets
     return _.uniqBy(currencies, (currencyId: SpacewalkPrimitivesCurrencyId) => currencyId.toHex());
   }, [vaults]);
 
   useEffect(() => {
-    if (!walletAccount || !wrappedCurrencies) return;
+    if (!walletAccount || !availableCurrencies) return;
 
     async function fetchBridgedTokens(address: string, asset: SpacewalkPrimitivesCurrencyId) {
       if (!api) return;
       return api.query.tokens.accounts(address, asset);
     }
 
-    wrappedCurrencies.forEach((currencyId) => {
+    availableCurrencies.forEach((currencyId) => {
       const walletAddress = ss58Format ? getAddressForFormat(walletAccount.address, ss58Format) : walletAccount.address;
       fetchBridgedTokens(walletAddress, currencyId).then((balance) => {
         setAccountTokenBalances((oldBalances) => {
-          const tokenId = currencyToString(currencyId);
-          if (tokenId && balance) {
-            oldBalances[tokenId] = balance;
+          try {
```

Comment:
I guess this try/catch is not needed, the full currencyToString() is wrapped in a try/catch block already, and right after we are asking for tokenId value making sure it is not null

Reply:
fixed it.